lightning/src/ln/functional_test_utils.rs

```diff
@@ -212,15 +212,15 @@ pub struct Node<'a, 'b: 'a, 'c: 'b> {
 	pub network_payment_count: Rc<RefCell<u8>>,
 	pub network_chan_count: Rc<RefCell<u32>>,
 	pub logger: &'c test_utils::TestLogger,
-	pub blocks: RefCell<Vec<(BlockHeader, u32)>>,
+	pub blocks: Arc<Mutex<Vec<(BlockHeader, u32)>>>,
```

Comment:
Why does this need to be an `Arc`?

Reply:
Ah, see https://github.com/rust-bitcoin/rust-lightning/pull/932#discussion_r641619700, but both the `Node` and the `TestBroadcaster` hold a reference to the same set of blocks.

Reply:
Right, but the objects which own the broadcaster are thread-safe, so we have to be :/